src/App.js

```diff
@@ -6,10 +6,18 @@ import { Routes, Route } from "react-router-dom";
 import { useState } from "react";
 import './App.css';
 import LandingPage from "./components/LandingPage";
+import { UserContext } from "./context/userContext";
 
 function App() {
   const [isLoading, setLoading] = useState(true);
```

Comment:
I'd recommend keeping the loading state in the relevant components. The way you have it here whenever the loading state is changed the whole App would be re-rendered. This a extra work that doesn't need to be done, for example if we're waiting for some comments to load we wouldn't need to re-render the NavBar for example :)